pkg/ccl/jobsccl/jobsprotectedtsccl/jobsprotectedtsccl_test: TestJobsProtectedTimestamp failed

pkg/ccl/jobsccl/jobsprotectedtsccl/jobsprotectedtsccl_test.TestJobsProtectedTimestamp [failed](https://tanzanite.cluster.engflow.com/invocations/default/2dc70570-b629-44ce-9ce0-0aff63dd5c62?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2pvYnNjY2wvam9ic3Byb3RlY3RlZHRzY2NsOmpvYnNwcm90ZWN0ZWR0c2NjbF90ZXN0) on release-24.2.0-rc @ [46c11ab2cf6fa6f6adf7abb99ef72f68e4d538f8](https://github.com/cockroachdb/cockroach/commits/46c11ab2cf6fa6f6adf7abb99ef72f68e4d538f8):


```
=== RUN   TestJobsProtectedTimestamp
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestJobsProtectedTimestamp1837014987
    test_log_scope.go:81: use -show-logs to present logs inline
    jobs_protected_ts_test.go:192: -- test log scope end --
--- FAIL: TestJobsProtectedTimestamp (54.45s)
=== RUN   TestJobsProtectedTimestamp/system-tenant
    jobs_protected_ts_test.go:132: 
        	Error Trace:	pkg/ccl/jobsccl/jobsprotectedtsccl/jobsprotectedtsccl_test/pkg/ccl/jobsccl/jobsprotectedtsccl/jobs_protected_ts_test.go:132
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1940
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:2027
        	            				github.com/cockroachdb/cockroach/pkg/kv/txn.go:1049
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:1103
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:1066
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:1041
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:2014
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1941
        	            				pkg/ccl/jobsccl/jobsprotectedtsccl/jobsprotectedtsccl_test/pkg/ccl/jobsccl/jobsprotectedtsccl/jobs_protected_ts_test.go:124
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            				github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:42
        	            				pkg/ccl/jobsccl/jobsprotectedtsccl/jobsprotectedtsccl_test/pkg/ccl/jobsccl/jobsprotectedtsccl/jobs_protected_ts_test.go:123
        	            				pkg/ccl/jobsccl/jobsprotectedtsccl/jobsprotectedtsccl_test/pkg/ccl/jobsccl/jobsprotectedtsccl/jobs_protected_ts_test.go:190
        	Error:      	Received unexpected error:
        	            	protected timestamp record does not exist
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts.init
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/protectedts.go:31
        	            	  | runtime.doInit1
        	            	  | 	GOROOT/src/runtime/proc.go:7206
        	            	  | runtime.doInit
        	            	  | 	GOROOT/src/runtime/proc.go:7173
        	            	  | runtime.main
        	            	  | 	GOROOT/src/runtime/proc.go:253
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) protected timestamp record does not exist
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestJobsProtectedTimestamp/system-tenant
    --- FAIL: TestJobsProtectedTimestamp/system-tenant (1.95s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/jobs @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestJobsProtectedTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40982

--- a/pkg/ccl/jobsccl/jobsprotectedtsccl/jobs_protected_ts_test.go
+++ b/pkg/ccl/jobsccl/jobsprotectedtsccl/jobs_protected_ts_test.go
@@ -68,48 +68,49 @@ func testJobsProtectedTimestamp(
 ) {
 	t.Helper()
 
-	mkJobRec := func() jobs.Record {
-		return jobs.Record{
-			Description: "testing",
-			Statements:  []string{"SELECT 1"},
-			Username:    username.RootUserName(),
-			Details: jobspb.SchemaChangeGCDetails{
-				Tables: []jobspb.SchemaChangeGCDetails_DroppedID{
-					{
-						ID:       42,
-						DropTime: clock.PhysicalNow(),
-					},
-				},
-			},
-			Progress:      jobspb.SchemaChangeGCProgress{},
-			DescriptorIDs: []descpb.ID{42},
-		}
-	}
 	insqlDB := execCfg.InternalDB
-	mkJobAndRecord := func() (j *jobs.Job, rec *ptpb.Record) {
+	mkJobAndRecord := func(f func(context.Context, isql.Txn, *jobs.Job) error) (rec *ptpb.Record) {
 		ts := clock.Now()
 		jobID := jr.MakeJobID()
-		require.NoError(t, insqlDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) (err error) {
-			if j, err = jr.CreateJobWithTxn(ctx, mkJobRec(), jobID, txn); err != nil {
+		require.NoError(t, insqlDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+			jobRec := jobs.Record{
+				Description: "testing",
+				Statements:  []string{"SELECT 1"},
+				Username:    username.RootUserName(),
+				Details: jobspb.SchemaChangeGCDetails{
+					Tables: []jobspb.SchemaChangeGCDetails_DroppedID{
+						{
+							ID:       42,
+							DropTime: clock.PhysicalNow(),
+						},
+					},
+				},
+				Progress:      jobspb.SchemaChangeGCProgress{},
+				DescriptorIDs: []descpb.ID{42},
+			}
+
+			j, err := jr.CreateJobWithTxn(ctx, jobRec, jobID, txn)
+			if err != nil {
 				return err
 			}
 			deprecatedSpansToProtect := roachpb.Spans{{Key: keys.MinKey, EndKey: keys.MaxKey}}
 			targetToProtect := ptpb.MakeClusterTarget()
 			rec = jobsprotectedts.MakeRecord(uuid.MakeV4(), int64(jobID), ts,
 				deprecatedSpansToProtect, jobsprotectedts.Jobs, targetToProtect)
-			return ptp.WithTxn(txn).Protect(ctx, rec)
+			if err := ptp.WithTxn(txn).Protect(ctx, rec); err != nil {
+				return err
+			}
+			return f(ctx, txn, j)
 		}))
-		return j, rec
+		return rec
 	}
-	jMovedToFailed, recMovedToFailed := mkJobAndRecord()
-	require.NoError(t, insqlDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
-		return jr.UnsafeFailed(ctx, txn, jMovedToFailed.ID(), io.ErrUnexpectedEOF)
-	}))
-	jFinished, recFinished := mkJobAndRecord()
-	require.NoError(t, insqlDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
-		return jr.Succeeded(ctx, txn, jFinished.ID())
-	}))
-	_, recRemains := mkJobAndRecord()
+	recMovedToFailed := mkJobAndRecord(func(ctx context.Context, txn isql.Txn, j *jobs.Job) error {
+		return jr.UnsafeFailed(ctx, txn, j.ID(), io.ErrUnexpectedEOF)
+	})
+	recFinished := mkJobAndRecord(func(ctx context.Context, txn isql.Txn, j *jobs.Job) error {
+		return jr.Succeeded(ctx, txn, j.ID())
+	})
+	recRemains := mkJobAndRecord(func(context.Context, isql.Txn, *jobs.Job) error { return nil })
 	ensureNotExists := func(ctx context.Context, txn isql.Txn, ptsID uuid.UUID) (err error) {
 		_, err = ptp.WithTxn(txn).GetRecord(ctx, ptsID)
 		if err == nil {
@@ -151,6 +152,13 @@ WHERE
 func TestJobsProtectedTimestamp(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	defer jobs.TestingRegisterConstructor(
+		jobspb.TypeSchemaChangeGC,
+		func(_ *jobs.Job, _ *cluster.Settings) jobs.Resumer {
+			return fakeResumer{}
+		},
+		jobs.UsesTenantCostControl)()
 
 	ctx := context.Background()
 	s0, db, _ := serverutils.StartServer(t, base.TestServerArgs{

